projects/graph/src/graph_demo.py

```diff
@@ -4,13 +4,43 @@
 Demonstration of Graph and BokehGraph functionality.
 """
 
+import random
 from sys import argv
+from draw import BokehGraph
+from graph import Graph, Vertex
 
 
-def main():
-    pass  # TODO
+
+def main(num_vertices=8, num_edges=8, draw_components=True):
+    """Build and show random graph."""
+    graph = Graph()
+    # Add appropriate number of vertices
+    for num in range(num_vertices):
+        # graph.add_vertex(Vertex(label=str(num)))
+     graph.add_vertex(Vertex(num))
+    # Add random edges between vertices
+    for _ in range(num_edges):
+       # vertices = sample(graph.vertices.keys(), 2)
+        # TODO check if edge already exists
+        rand_vert = int(random.random() * num_vertices)
+        rand_edge = int(random.random() * num_edges)
+        print(rand_vert, rand_edge)
+        graph.add_edge(vertices[0], vertices[1])
```

Comment:
Would this need to be num_vertices instead of vertices? Let me know if you want help with debugging this. It looks like a great start on getting the random graph to draw but the interpreter is getting hung up on this line.